### PR TITLE
Refresh `EndpointConfig` request headers

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Refresh headers used in requests (e.g. action server requests) made by `EndpointConfig` using its `headers` attribute. 
- This is particularly relevant when the tracing context is propagated from `rasa-plus` to the action server.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/endpoints.py [hunk 1]
```diff
@@ -149,6 +149,9 @@ async def request(
             headers.update(kwargs["headers"])
```
Comment (vcidst) [inline rasa/utils/endpoints.py:149]: Will the changes done to the headers here be "undone" by your changes?

Comment (vcidst) [reply rasa/utils/endpoints.py:149]: There is a possibility that the similar keys get overwritten here. Do you have any ideas what can be done to minimise this risk, perhaps only add the tracing related header key?

Comment (ancalita) [reply rasa/utils/endpoints.py:149]: It depends what could be in the `self.headers`, in the tracing scenarios it only contains the `traceparent` key-value pair. I've had a look at where this `EndpointConfig.request()` method is used in the codebase, and only in the `rasa interactive` code it is used in two places with `headers` keyword argument with value `{"Accept": "application/json"}` , let me know if you're interested and I can share the links.

Comment (ancalita) [reply rasa/utils/endpoints.py:149]: > only add the tracing related header key

yep sure, we can restrict to this only, however I think the risk is low for overwriting keys based on the code usage I could find of `headers` keyword argument (see reply above). Let me know before making the commit if I should go ahead?

Comment (vcidst) [reply rasa/utils/endpoints.py:149]: If you've looked into it already then it's cool. I'm good with it

**Review by vcidst — APPROVED** [review]

**Pull request merged**